File: frontend/src/types/outfit.ts
/**
 * 穿搭推荐类型定义
 * 与后端 `app/schemas/outfit.py` 对齐
 */

/** 穿搭单品 */
export interface OutfitItem {
  name: string
  category?: string
  note?: string
  gender?: 'female' | 'male' | 'unisex' | string
}

/** 穿搭推荐 */
export interface OutfitRecommendation {
  id: string
  trip_id: string
  scene: string
  season: string
  style: string
  items: OutfitItem[]
  tips: string | null
  images: string[]
  created_at: string
  updated_at: string
}

/** 穿搭推荐创建请求 */
export interface OutfitRecommendationCreate {
  trip_id?: string | null
  scene: string
  season: string
  style: string
  items: OutfitItem[]
  tips?: string | null
  images?: string[]
}

/** 穿搭推荐更新请求 */
export interface OutfitRecommendationUpdate {
  scene?: string
  season?: string
  style?: string
  items?: OutfitItem[]
  tips?: string | null
  images?: string[]
}

/** AI 穿搭预览图生成结果 */
export interface OutfitPreviewImageResult {
  outfit: OutfitRecommendation
  image_url: string | null
  provider: string
  generated: boolean
  message: string | null
}

File: backend/app/schemas/outfit.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.domain import ORMModel


class OutfitRecommendationCreate(BaseModel):
    trip_id: UUID | None = None
    scene: str
    season: str
    style: str
    items: list[dict] = Field(default_factory=list)
    tips: str | None = None
    images: list[str] = Field(default_factory=list)


class OutfitRecommendationUpdate(BaseModel):
    scene: str | None = None
    season: str | None = None
    style: str | None = None
    items: list[dict] | None = None
    tips: str | None = None
    images: list[str] | None = None


class OutfitPreviewImageRequest(BaseModel):
    force: bool = False


class OutfitRecommendationRead(ORMModel):
    id: UUID
    trip_id: UUID
    scene: str
    season: str
    style: str
    items: list[dict]
    tips: str | None
    images: list[str]
    created_at: datetime
    updated_at: datetime


class OutfitPreviewImageRead(BaseModel):
    outfit: OutfitRecommendationRead
    image_url: str | None = None
    provider: str
    generated: bool
    message: str | None = None
